dartagnan/src/main/java/com/dat3m/dartagnan/expression/IConst.java

```diff
@@ -18,12 +16,7 @@ public abstract class IConst extends IExpr implements LastValueInterface {
      */
     public abstract BigInteger getValue();
 
-	@Override
-    public Formula toIntFormula(Event e, FormulaManager m) {
-		return toIntFormula(m);
-	}
-
-	public Formula toIntFormula(FormulaManager fmgr) {
+    public Formula toIntFormula(FormulaManager fmgr) {
 		return getPrecision() > 0
 				? fmgr.getBitvectorFormulaManager().makeBitvector(getPrecision(),getValue())
 				: fmgr.getIntegerFormulaManager().makeNumber(getValue());
```

Comment:
`IConst.toIntFormula(FormulaManager)` is still used by `getLastValueExpr(FormulaManager)`.  I could let `ExpressionEncoder` also accept `event = null` and then let it cover this method, as well.  This could also be a step into merging `AbstractAssert` into `ExprInterface`.  Just trying to locally evaluate `Location` will throw an exception.

Reply:
There is also still `Register.toIntFormulaResult(Event, FormulaManager)`.

Reply:
Update: `LastValueInterface` has been removed and there is no usage of `org.sosy_lab.java_smt.api` inside `com.dat3m.dartagnan.expression` or any implementation of `ExprInterface` anymore.